### PR TITLE
Add organization admin crud events

### LINKED ISSUE #30180: Keycloak Organizations
### Description

This top-level epic is about the planned work to deliver the first release of the Keycloak Organization feature.

Keycloak Organization is a feature that leverages and builds on top of the Identity and Access Management (IAM) capabilities of Keycloak to address Customer Identity and Access Management (CIAM) with a focus on Business-to-Business (B2B) use cases.

                                        +-----------+                  
                                        |           |                  
                            +-----------+  Customer +-----------+      
                            |           |           |           |      
    +----------+      +-----+----+      |           |      +----+-----+
    |          |      |          |      +-----------+      |          |
    | End-User +------+  Realm   |                         | End-User |
    |          |      |          |                         |          |
    |          |      |          |                         |          |
    +----------+      +-----+----+                         +----+-----+
                            |           +-----------+           |      
                            |           |           |           |      
                            +-----------+  Partner  +-----------+      
                                        |           |                  
                                        |           |                  
                                        +-----------+                       

In summary, this feature will allow a realm to integrate with third-party entities like customers and business partners so that their identities and how they access protected resources at the realm level are segregated and managed within the scope of the organization they belong to. 

As a result, a realm will be able to provide a different experience when onboarding customer identities or identities from a business partner and mechanisms to secure and manage how they interact with a realm and how they access protected resources from the clients available from a realm. Although not the main focus, it should also be possible to leverage this feature if you just want a better segregation between users, clients, and how they are managed and authenticate to a realm.

The main set of capabilities provided by Keycloak Organizations are:

* * [x] **Manage Organizations**
* * [ ] **Manage Organization Members**
* * * [x] Organization Member Self-Registration based on Identity Brokering
* * * [x] Organization Member Registration based on Invitation Links
* * * [ ] Joining Multiple Organizations 
* * [x] **Mapping organization metadata in tokens** 
* * [ ] **Manage Organization Roles**
* * [ ] **Manage Organization Groups**
* * [ ] **Manage Authentication within the Scope of an Organization**
* * * [ ] Authentication based on the Organization Authentication Policies**
* * [ ] **Manage Clients and Service Accounts**
* * [ ] **Administrative Authorization**
* * * [ ] **RBAC**
* * * [ ] **Fine-Grained** 
* * [ ] **Organization Self-Service**

The planned release dates for Keycloak Organization are:

| Release  | State |
| ------------- | ------------- |
| 25.0.0  | Preview  |
| 26.0.0  | Supported  |

For more details about the release dates, see https://github.com/keycloak/keycloak/milestones.

### Discussion

https://github.com/keycloak/keycloak/discussions/23948

### Issues

- [First Release of Keycloak Organizations](https://github.com/keycloak/keycloak/issues/28609)

### Motivation

Keycloak is a well-known open-source IAM solution and we want to leverage now its capabilities to enable CIAM with a focus on B2B use cases.

### LINKED ISSUE #31421: Add Events for Organization Creation and Member Assignment
### Description

It would be nice to be able to listen for organization related events in a custom spi interface.

### Discussion

#30180

### Motivation

This feature will allow me to call my internal API in order to create that organization in a database and assign users to them

### Details

_No response_

### PR DESCRIPTION
**Pull request by mzvankovich**
Add organization admin crud events

Related #31421, #30180

### CODE REVIEW

**Comment by ahus1** [comment]
@pedroigor - this might be of interest for your team.

**Review by ssilvert — APPROVED** [review]
UI part approved.

**Comment by ahus1** [comment]
@mzvankovich - While this is in review by a maintainer, there is no need to rebase your PR unless you see conflicts in your PR. A maintainer will automatically rebase it when it is merged to main. This avoid using CI resources unnecessarily. Thanks!

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java
Comment (mzvankovich) [reply server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java:1312]: @sguilhen thank you for the review. Ok. I'll move that mapping into `Organizations.java`.

**Comment by sguilhen** [comment]
@mzvankovich Actually, put this on hold for a moment - I'm refactoring a few things to properly export/import the orgs (the whole thing about preserving the ids), and I'm moving some things around because the `Organizations` class is not reachable from the export utils atm. So I'll prob move the code the `ModelToRepresentation` like you did here.

**Comment by sguilhen** [comment]
@mzvankovich I am not able to push changes to your PR, so I've cherry-picked your changes and opened another PR to replace this one: https://github.com/keycloak/keycloak/pull/33281

**Pull request closed without merging**
